packages/rest/test/integration/rest.server.integration.ts

```diff
@@ -159,6 +172,110 @@ describe('RestServer (integration)', () => {
   });
 
+  it('exposes "GET /openapi.json" endpoint with openApiSpec opts', async () => {
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        openApiSpec: {
+          info: {
+            title: 'My Application',
+            version: '1.0.0',
+          },
+          servers: [{url: 'http://127.0.0.1:8080'}],
+        },
+      },
+    });
+    const greetSpec = {
+      responses: {
+        200: {
+          content: {'text/plain': {schema: {type: 'string'}}},
+          description: 'greeting of the day',
+        },
+      },
+    };
+    server.route(new Route('get', '/greet', greetSpec, function greet() {}));
+
+    const response = await createClientForHandler(server.requestHandler).get(
+      '/openapi.json',
+    );
+    expect(response.body).to.containDeep({
+      openapi: '3.0.0',
+      info: {
+        title: 'My Application',
+        version: '1.0.0',
+      },
+      servers: [{url: 'http://127.0.0.1:8080'}],
+      paths: {
+        '/greet': {
+          get: {
+            responses: {
+              '200': {
+                content: {
+                  'text/plain': {
+                    schema: {type: 'string'},
+                  },
+                },
+                description: 'greeting of the day',
+              },
+            },
+          },
+        },
+      },
+    });
+    expect(response.get('Access-Control-Allow-Origin')).to.equal('*');
+    expect(response.get('Access-Control-Allow-Credentials')).to.equal('true');
+  });
+
+  it('exposes "GET /openapi.json" endpoint with setServersFromRequest', async () => {
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        apiExplorer: {
+          setServersFromRequest: true,
+        },
+      },
+    });
+    const greetSpec = {
+      responses: {
+        200: {
+          content: {'text/plain': {schema: {type: 'string'}}},
+          description: 'greeting of the day',
+        },
+      },
+    };
+    server.route(new Route('get', '/greet', greetSpec, function greet() {}));
+
+    const response = await createClientForHandler(server.requestHandler).get(
+      '/openapi.json',
+    );
+    expect(response.body).to.containDeep({
+      openapi: '3.0.0',
+      info: {
+        title: 'LoopBack Application',
+        version: '1.0.0',
+      },
+      paths: {
+        '/greet': {
+          get: {
+            responses: {
+              '200': {
+                content: {
+                  'text/plain': {
+                    schema: {type: 'string'},
+                  },
+                },
+                description: 'greeting of the day',
+              },
+            },
+          },
+        },
+      },
+    });
+    expect(response.body.servers[0].url).to.match(/http:\/\/127.0.0.1\:\d+/);
+    expect(response.get('Access-Control-Allow-Origin')).to.equal('*');
+    expect(response.get('Access-Control-Allow-Credentials')).to.equal('true');
```

Comment:
Ditto. 

IIUC, this test is verifying how `rest.openApiSpec` config is applied. There is no need to check CORS headers in such case, they have been already verified by previous tests.

Please remove these two assertions.